aana/core/models/image_chat.py

```diff
@@ -0,0 +1,139 @@
+from typing import Annotated, Any, Literal
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from aana.core.models.image import Image
+
+Role = Literal["system", "user", "assistant"]
+
+
+class TextContent(BaseModel):
+    """Text content for a chat message.
+
+    Attributes:
+        type (Literal["text"]): the type of the content, always "text"
+        text (str): the text of the message
+    """
+
+    type: Literal["text"] = "text"
+    text: str
+
+
+class ImageContent(BaseModel):
+    """Image content for a chat message.
+
+    Attributes:
+        type (Literal["image"]): the type of the content, always "image"
+        image (Image): the image
+    """
+
+    type: Literal["image"] = "image"
+    image: Image
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+Content = Annotated[
+    TextContent | ImageContent,
+    Field(description="The content of the message, either text or image."),
+]
+
+
+class ImageChatMessage(BaseModel):
+    """A chat message with image support.
+
+    Attributes:
+        content (list[Content]): the content of the message
+        role (Role): the role of the message
+    """
+
+    content: list[Content]
+    role: Role
+    model_config = ConfigDict(
+        json_schema_extra={"description": "A chat message with image support."}
+    )
+
+
+class ImageChatDialog(BaseModel):
+    """A chat dialog with image support.
+
+    Attributes:
+        messages (list[ImageChatMessage]): the list of messages
+    """
+
+    messages: list[ImageChatMessage]
+    model_config = ConfigDict(
+        json_schema_extra={"description": "A chat dialog with image support."}
+    )
+
+    @classmethod
+    def from_list(cls, messages: list[dict[str, Any]]) -> "ImageChatDialog":
+        """Create an ImageChatDialog from a list of messages.
+
+        Args:
+            messages (list[dict[str, str]]): the list of messages
+
+        Returns:
+            ImageChatDialog: the chat dialog
+        
+        Example:
+        ```
+            messages = [
```

Comment:
Remove identation for the example. It adds unnecessary identation in the code block when docs are rendered.